emcc.py

```diff
@@ -369,7 +369,11 @@ def get_clang_flags(user_args):
   if settings.RELOCATABLE and '-fPIC' not in user_args:
     flags.append('-fPIC')
 
-  if settings.RELOCATABLE or settings.LINKABLE or '-fPIC' in user_args:
+  # Whether this compile command is emitting something that will only be used in
+  # a whole-program link, that is, without dynamic linking being possible.
+  for_whole_program_link = not settings.RELOCATABLE and not settings.LINKABLE and '-fPIC' not in user_args
```

Comment:
How about inverting this condition and the name of this variable.

e.g.

```
dylink_supported = settings.RELOCATABLE or settings.LINKABLE or '-fPIC' in user_args
```

Then I believe the '-fwhole-program-vtables' should be added only when this is *not* true... so it would not be in the same block as `-fvisibility=default' (which is for relocatable output).

Reply:
Another name for this might be "closed_world"?   I'm not sure its any better though.  

LLVM itself calls this this concept "dso_local" internally, FWIW

Reply:
I like `dylink_supported`, good idea, done.